src/AutorestSwift/Code Generator/Manager.swift

```diff
@@ -33,56 +33,68 @@ struct StringDecodeKeys: ExtensionStringDecodeKeys {
 
 /// Handles the configuration and orchestrations of the code generation process
 class Manager {
-    internal enum InvocationMode {
-        case autorest
-        case commandLine
-    }
+    static var shared = Manager()
 
     // MARK: Properties
 
-    let inputString: String
-
-    let destinationRootUrl: URL
+    var configured: Bool
+    var config: ManagerConfig
+    var inputString: String
+    var destinationRootUrl: URL!
     var packageUrl: URL?
```

Comment:
Should destinationRootUrl and packageUrl part of ManagerConfig as well?

Reply:
Probably